### DISCUSSION TITLE
IPv6 in IPv4 tunnel. Can't interact with local IPv6 LAN or myself

### DISCUSSION THREAD

**Discussion by haraldhh** [strongswan/strongswan#1343] [Q&A] [answered]
This might be by design, but I have to ask. I have a IPv6 in IPv4 tunnel set up like in https://www.strongswan.org/testing/testresults/ipv6-stroke/net2net-ip6-in-ip4-ikev2/. Everything works. 

I did have to add "routing_table = 254 # main" to charon in strongswan.conf for the clients on my local IPv6 LAN to be routed. 

I am however unable to ping6 hosts on the same IPv6 subnet as the moon gateway. I am also not able to contact myself on the IPv6 address on the moon gateway (that is, the IPv6 address that is assigned to the gateway itself). 

Pinging a host on the local IPv6 LAN from the moon gateway says "Time exceeded: Hop limit". 

Is it possible to configure IPsec so that I am able to be both a part of the local IPv6 LAN *and' act as the "moon" IPsec router?

the "moon" gw:
```
# ipsec.conf - strongSwan IPsec configuration file

# basic configuration

config setup
	# strictcrlpolicy=yes
	# uniqueids = no
  charondebug="ike 1, knl 1, cfg 1"
# Add connections here.


conn %default
	ikelifetime=60m
	keylife=20m
	rekeymargin=3m
	keyingtries=1
	keyexchange=ikev2
	mobike=no

conn workplace
	left=%any4
	leftsubnet=2001:xxx:xxx:dead::/64
	leftcert=harald.pem
	leftid=@harald
	leftfirewall=yes
	right=193.xxx.xxx.2
	rightsubnet=::/0
	rightid="C=FI, O=org, CN=vpn.domain.com"
	auto=add

conn ndp-ns
   right=::1 # so this connection does not get used for other purposes
    leftsubnet=::/0[ipv6-icmp/135]
    rightsubnet=::/0[ipv6-icmp/135]
    type=passthrough
    auto=route

conn ndp-na
    right=::1 # so this connection does not get used for other purposes
    leftsubnet=::/0[ipv6-icmp/136]
    rightsubnet=::/0[ipv6-icmp/136]
    type=passthrough
    auto=route
```

the "sun" gw:
```
# ipsec.conf - strongSwan IPsec configuration file

# basic configuration

config setup
	# strictcrlpolicy=yes
	# uniqueids = no

conn %default
	ikelifetime=60m
	keylife=20m
	rekeymargin=3m
	keyingtries=1
	keyexchange=ikev2
	mobike=no

conn harald
	left=193.xxx.xx.2
	leftsubnet=::/0
	leftcert=vpn.domain.com-cert.pem
	leftid=@vpn.domain.com
	leftfirewall=yes
	right=%any
	rightsubnet=2001:xxx:xxx:dead::/64
	rightid="C=FI, O=org, CN=harald"
	auto=start
```

I have assigned 2001:xxx:xxx:dead::1/64 to my "moon" gw. I am unabled to 'telnet 2001:xxx:xxx:dead::1 22' from 2001:xxx:xxx:dead::1 itself. I can't ping another host in the 2001:xxx:xxx:dead::/64 from 2001:xxx:xxx:dead::1. 

Really grateful for Your help

**Comment by haraldhh** [comment]
Could the problem be routing? That my routing table thinks that connections from and to 2001:xxx:xxx:dead::1 should be on the lo interface, and IPsec says no?

```
$ ip ro get 2001:xxx:xxx:dead::1
local 2001:xxx:xxx:dead::1 from :: dev lo table local proto kernel src 2001:xxx:xxx:dead::1 metric 0 pref medium
```

**Comment by tobiasbrunner** [comment, marked as answer]
With a remote TS of `::/0`  you'll have to install a [bypass policy](https://docs.strongswan.org/docs/5.9/config/quickstart.html#_passthroughbypass_policies) for your local subnet. With the default value for `charon.routing_table` this will also install a throw route in table 220 so the main routing table will be consulted for that subnet.

**Comment by haraldhh** [comment]
Very much thank You Sir! 

I put this in ipsec.conf on "moon";
```
conn localnet
    left=::1
    leftsubnet=2001:xxx:xxx:dead::/64
    rightsubnet=2001:xxx:xxx:dead::/64
    authby=never
    type=passthrough
    auto=route
```

This fixed it. Thank You.
